scripts/utils/cliUtils.test.ts

```diff
@@ -0,0 +1,37 @@
+import { InvalidArgumentError } from '@commander-js/extra-typings'
+import { testVariadicArgs } from './cliUtils'
+
+const sampleTesterFunctionIsString = (str: string) => {
+  return typeof str === 'string'
+}
+
+describe('test variadic args', () => {
+  it('tester function is called, it returns false. test error', async () => {
+    try {
+      testVariadicArgs(
+        123 as unknown as string,
+        '123',
+        sampleTesterFunctionIsString
+      )
+    } catch (error) {
+      expect(error).toBeInstanceOf(InvalidArgumentError)
+      expect((error as Error).message).toBe(
+        'Value provided does not match supported format'
+      )
+    }
+  })
```

Comment:
This is difficult to read/requires high mental load to unpick. 

Please extract the three arguments into clearly named variables e.g. 
    const invalidNumberAsString = 123 as unknown as string;
    const validString = '123';
    const invalidFunction = sampleTesterFunctionIsString

    try {
      testVariadicArgs(invalidNumberAsString, validString, invalidFunction); } etc...